allenact/embodiedai/models/visual_nav_models.py

```diff
@@ -0,0 +1,210 @@
+        self.beliefs_fusion = beliefs_fusion
+        self.auxiliary_uuids = auxiliary_uuids
+        if isinstance(self.auxiliary_uuids, list) and len(self.auxiliary_uuids) == 0:
+            self.auxiliary_uuids = None
+
+    def create_state_encoders(
+        self,
+        obs_embed_size: int,
+        prev_action_embed_size: int,
+        num_rnn_layers: int,
+        rnn_type: str,
+        add_prev_actions: bool,
+        trainable_masked_hidden_state=False,
+    ):
+        rnn_input_size = obs_embed_size
+        self.prev_action_embedder = FeatureEmbedding(
+            input_size=self.action_space.n,
+            output_size=prev_action_embed_size if add_prev_actions else 0,
+        )
+        if add_prev_actions:
+            rnn_input_size += prev_action_embed_size
+
+        state_encoders = OrderedDict()  # perserve insertion order in py3.6
+        if self.multiple_beliefs:  # multiple belief model
+            for aux_uuid in self.auxiliary_uuids:
+                state_encoders[aux_uuid] = RNNStateEncoder(
+                    rnn_input_size,
+                    self._hidden_size,
+                    num_layers=num_rnn_layers,
+                    rnn_type=rnn_type,
+                    trainable_masked_hidden_state=trainable_masked_hidden_state,
+                )
+            # create fusion model
+            self.fusion_model = FusionModels[self.beliefs_fusion](
+                hidden_size=self._hidden_size,
+                obs_embed_size=obs_embed_size,
+                num_tasks=len(self.auxiliary_uuids),
+            )
+
+        else:  # single belief model
+            state_encoders["single_belief"] = RNNStateEncoder(
+                rnn_input_size,
+                self._hidden_size,
+                num_layers=num_rnn_layers,
+                rnn_type=rnn_type,
+                trainable_masked_hidden_state=trainable_masked_hidden_state,
+            )
+
+        self.state_encoders = nn.ModuleDict(state_encoders)
+
+        self.belief_names = list(self.state_encoders.keys())
+
+        get_logger().info(
+            "there are {} belief models: {}".format(
+                len(self.belief_names), self.belief_names
+            )
+        )
+
+    def create_actorcritic_head(self):
+        self.actor = LinearActorHead(self._hidden_size, self.action_space.n)
+        self.critic = LinearCriticHead(self._hidden_size)
+
+    def create_aux_models(self, obs_embed_size: int, action_embed_size: int):
+        if self.auxiliary_uuids is None:
+            return
+        aux_models = OrderedDict()
+        for aux_uuid in self.auxiliary_uuids:
+            aux_models[aux_uuid] = AuxiliaryModel(
+                aux_uuid=aux_uuid,
+                action_dim=self.action_space.n,
+                obs_embed_dim=obs_embed_size,
+                belief_dim=self._hidden_size,
+                action_embed_size=action_embed_size,
+            )
+
+        self.aux_models = nn.ModuleDict(aux_models)
```

Comment:
Can you add a `self.aux_models = None` in the `__init__` so that this instance variable is not created inside another function?